### PR TITLE
[Tests-Only] Expand API tests for addUser, createSubAdmin, deleteUser, disableUser, enableUser and editUser

### PR DESCRIPTION
**Pull request by saw-jan**
## Description
expands apiProvisioning tests for ocs v1 and v2 in:
- addUser
- deleteUser
- editUser
- disableUser
- enableUser
- createSubAdmin

## Related Issue

- part of https://github.com/owncloud/core/issues/31533

## How Has This Been Tested?
- test environment: local

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:
- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [ ] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

**Comment by phil-davis** [comment]
Something failed in drone https://drone.owncloud.com/owncloud/core/28333/58/13
But sadly the log file output was not fully flushed, so we cannot see exactly what failed.
I have restarted drone CI.

**Comment by phil-davis** [comment]
https://drone.owncloud.com/owncloud/core/28335/58/13
```
--- Failed scenarios:

    /drone/src/tests/acceptance/features/apiProvisioning-v2/addUser.feature:118
    /drone/src/tests/acceptance/features/apiProvisioning-v2/addUser.feature:119
    /drone/src/tests/acceptance/features/apiProvisioning-v2/addUser.feature:120
    /drone/src/tests/acceptance/features/apiProvisioning-v2/addUser.feature:121
    /drone/src/tests/acceptance/features/apiProvisioning-v2/addUser.feature:122
    /drone/src/tests/acceptance/features/apiProvisioning-v2/addUser.feature:123

150 scenarios (144 passed, 6 failed)
1137 steps (1113 passed, 6 failed, 18 skipped)
47m10.10s (17.12Mb)
```

This time you can see what failed.

File: tests/acceptance/features/apiProvisioning-v1/editUser.feature [hunk 1]
```diff
@@ -97,6 +109,43 @@ Feature: edit users
...
+    And group "new-group" has been created
+    And user "another-subadmin" has been made a subadmin of group "new-group"
+    And user "subadmin" has been made a subadmin of group "new-group"
+    When user "subadmin" changes the quota of user "another-subadmin" to "12MB" using the provisioning API
+    Then the OCS status code should be "997"
+    And the HTTP status code should be "401"
+    When user "subadmin" changes the email of user "another-subadmin" to "brand-new-user@example.com" using the provisioning API
+    Then the OCS status code should be "997"
+    And the HTTP status code should be "401"
+    And user "subadmin" changes the display of user "another-subadmin" to "Anne Brown" using the provisioning API
```
Comment (Talank) [inline tests/acceptance/features/apiProvisioning-v1/editUser.feature:142]:
```suggestion
    When user "subadmin" changes the display of user "another-subadmin" to "Anne Brown" using the provisioning API
```
This looks like a when step

File: tests/acceptance/features/apiProvisioning-v1/editUser.feature [hunk 2]
```diff
@@ -97,6 +109,43 @@ Feature: edit users
...
+    Then the OCS status code should be "997"
+    And the HTTP status code should be "401"
+    Then the display name of user "another-subadmin" should be "Regular User"
```
Comment (Talank) [inline tests/acceptance/features/apiProvisioning-v1/editUser.feature:145]:
```suggestion
    And the display name of user "another-subadmin" should be "Regular User"
```

File: tests/acceptance/features/apiProvisioning-v2/editUser.feature [hunk 3]
```diff
@@ -97,6 +109,47 @@ Feature: edit users
...
+    And user "subadmin" has been made a subadmin of group "new-group"
+    When user "subadmin" changes the quota of user "another-subadmin" to "12MB" using the provisioning API
+    # Then the OCS status code should be "401"
+    Then the OCS status code should be "997"
+    And the HTTP status code should be "401"
+    When user "subadmin" changes the email of user "another-subadmin" to "brand-new-user@example.com" using the provisioning API
+    # Then the OCS status code should be "401"
+    Then the OCS status code should be "997"
+    And the HTTP status code should be "401"
+    And user "subadmin" changes the display of user "another-subadmin" to "Anne Brown" using the provisioning API
```
Comment (Talank) [inline tests/acceptance/features/apiProvisioning-v2/editUser.feature:145]:
```suggestion
    When user "subadmin" changes the display of user "another-subadmin" to "Anne Brown" using the provisioning API
```

File: tests/acceptance/features/apiProvisioning-v2/editUser.feature [hunk 4]
```diff
@@ -97,6 +109,47 @@ Feature: edit users
...
+    # Then the OCS status code should be "401"
+    Then the OCS status code should be "997"
+    And the HTTP status code should be "401"
+    Then the display name of user "another-subadmin" should be "Regular User"
```
Comment (Talank) [inline tests/acceptance/features/apiProvisioning-v2/editUser.feature:149]:
```suggestion
    And the display name of user "another-subadmin" should be "Regular User"
```

**Review by dpakach — COMMENTED** [review]
Everything looks good but try adding tests by not changing the line number of existing tests.

**Review by Talank — COMMENTED** [review]
LGTM

File: tests/acceptance/features/apiProvisioning-v2/enableUser.feature [hunk 5]
```diff
@@ -89,3 +89,69 @@ Feature: enable user
...
+    And group "brand-new-group" has been created
+    And user "Alice" has been added to group "brand-new-group"
+    And user "subadmin" has been made a subadmin of group "brand-new-group"
+    And user "Alice" has been disabled
+    When user "subadmin" tries to enable user "Alice" using the provisioning API
+    Then the OCS status code should be "200"
+    And the HTTP status code should be "200"
+    And user "Alice" should be enabled
+
+  Scenario: subadmin should not be to enable user not in their group
```
Comment (individual-it) [inline tests/acceptance/features/apiProvisioning-v2/enableUser.feature:117]:
```suggestion
  Scenario: subadmin should not be able to enable user not in their group
```

**Review by individual-it — DISMISSED** [review]

**Review by individual-it — APPROVED** [review]

**Pull request merged**